apps/test/unit/code-studio/initApp/loadAppTest.js

```diff
@@ -153,4 +155,56 @@ describe('loadApp.js', () => {
       done();
     });
   });
+
+  describe('project level share images', () => {
+    const BLANK_PNG_PIXEL = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=';
+
+    beforeEach(() => {
+      sinon.spy(imageUtils, 'dataURIToFramedBlob');
+      sinon.stub(files, 'putFile');
+      appOptions.level.isProjectLevel = true;
+      appOptions.level.edit_blocks = false;
+    });
+
+    afterEach(() => {
+      files.putFile.restore();
+      imageUtils.dataURIToFramedBlob.restore();
+    });
+
+    it('uploads a share image for a non-droplet project (instead of writing the level)', (done) => {
+      files.putFile.callsFake((name, blob) => {
+        expect(writtenLevelId).to.be.undefined;
+        expect(name).to.equal('_share_image.png');
+        expect(blob).to.have.property('type', 'image/png');
+        expect(blob).to.have.property('size', 523181);
+        done();
+      });
+
+      setupApp(appOptions);
+      appOptions.onAttempt({image: BLANK_PNG_PIXEL});
+
+      // dataURIToFramedBlob gets called synchronously, eventually calls putFile.
+      expect(imageUtils.dataURIToFramedBlob).to.have.been.calledOnce;
```

Comment:
These tests have a little too much knowledge of the implementation here, but initApp is tricky and I think we're better off having brittle tests here than no tests at all.